src/main/java/com/conveyal/r5/streets/LinkedPointSet.java

```diff
@@ -497,36 +520,88 @@ public PointSetTimes eval (
             if (edges[p] == -1) {
                 return true; // Continue to next iteration.
             }
-            edge.seek(edges[p]);
-            int t1 = Integer.MAX_VALUE;
-            int t2 = Integer.MAX_VALUE;
-            // TODO this is not strictly correct when there are turn restrictions onto the edge this is linked to
-            if (distanceTableToVertices.containsKey(edge.getFromVertex())) {
-                t1 = distanceTableToVertices.get(edge.getFromVertex()) + distances0_mm[p] + distancesToEdge_mm[p];
+
+            if (egressArea != null && !GeometryUtils.containsPoint(egressArea, pointSet.getLon(p), pointSet.getLat(p))) {
+                return true; // Point is outside supplied area, continue to next iteration.
             }
-            if (distanceTableToVertices.containsKey(edge.getToVertex())) {
-                t2 = distanceTableToVertices.get(edge.getToVertex()) + distances1_mm[p] + distancesToEdge_mm[p];
+
+            edge.seek(edges[p]);
+
+            int cost = Integer.MAX_VALUE;
+
+            if (routingVariable == RoutingVariable.DISTANCE_MILLIMETERS) {
+                cost = distanceToPoint(costTableToVertices, edge, p);
+            } else if (routingVariable == RoutingVariable.DURATION_SECONDS) {
+                int time0 = costTableToVertices.get(edge.getFromVertex());
+                int time1 = costTableToVertices.get(edge.getFromVertex());
+                if (time0 == 0 && time1 == 0) {
+                    return true; // Edge unreachable, continue iteration. Should we update sr.getReachedVertices to use
+                    // a different noEntryValue?
+                }
+                int onStreetSpeed = (int) edge.getCarSpeedMetersPerSecond() * 1000;
+                cost = timeToPoint(time0, time1, onStreetSpeed, OFF_STREET_SPEED_MILLIMETERS_PER_SECOND, p);
             }
-            int t = Math.min(t1, t2);
-            if (t != Integer.MAX_VALUE) {
-                if (t < distanceToPoint.get(p)) {
-                    distanceToPoint.put(p, t);
+
+            if (cost != Integer.MAX_VALUE) {
+                if (cost < costToPoint.get(p)) { // When is this false?
```

Comment:
Response posted in Slack: The method initializes distanceToPoint to an empty int-int map, then iterates over all the points in the envelope and saves one distance to each point in the map, keyed on the point number p. It does look like this conditional could only ever be hit once for a given value of p, and the default value before we write to the map is the max integer, so yes I think it would always be true. It looks like maybe the method was adapted from a pre-existing method that iterated over many candidates and kept the best one. The loop that associates the points with edges would work that way.